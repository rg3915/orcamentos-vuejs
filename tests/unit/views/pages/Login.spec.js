import Vue from 'vue'
import { shallowMount } from '@vue/test-utils'
import CoreuiVue from '@coreui/vue'
import Login from '@/views/pages/Login'

Vue.use(CoreuiVue)

describe('Login.vue', () => {
  it('has a name', () => {
    expect(Login.name).toBe('Login')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Login)
    expect(wrapper.vm).toBeTruthy()
  })
  it('is Login', () => {
    const wrapper = shallowMount(Login)
    expect(wrapper.findComponent(Login)).toBeTruthy()
  })
  it('should render correct content', () => {
    const wrapper = shallowMount(Login)
    expect(wrapper.find('h1').text()).toMatch('Login')
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Login)
    expect(wrapper.element).toMatchSnapshot()
  })
})
