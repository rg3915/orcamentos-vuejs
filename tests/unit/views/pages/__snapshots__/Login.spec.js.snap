// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login.vue renders correctly 1`] = `
<div
  class="c-app flex-row align-items-center"
>
  <ccontainer-stub
    tag="div"
  >
    <crow-stub
      class="justify-content-center"
      gutters="true"
      tag="div"
    >
      <ccol-stub
        md="8"
        tag="div"
      >
        <ccardgroup-stub
          tag="div"
        >
          <ccard-stub
            class="p-4"
          >
            <ccardbody-stub>
              <cform-stub>
                <h1>
                  Login
                </h1>
                 
                <p
                  class="text-muted"
                >
                  Sign In to your account
                </p>
                 
                <cinput-stub
                  autocomplete="username email"
                  lazy="400"
                  placeholder="Username"
                  type="text"
                />
                 
                <cinput-stub
                  autocomplete="curent-password"
                  lazy="400"
                  placeholder="Password"
                  type="password"
                />
                 
                <crow-stub
                  gutters="true"
                  tag="div"
                >
                  <ccol-stub
                    class="text-left"
                    col="6"
                    tag="div"
                  >
                    <cbutton-stub
                      activeclass="router-link-active"
                      class="px-4"
                      color="primary"
                      event="click"
                      exactactiveclass="router-link-exact-active"
                      routertag="a"
                      target="_self"
                      type="button"
                    >
                      Login
                    </cbutton-stub>
                  </ccol-stub>
                   
                  <ccol-stub
                    class="text-right"
                    col="6"
                    tag="div"
                  >
                    <cbutton-stub
                      activeclass="router-link-active"
                      class="px-0"
                      color="link"
                      event="click"
                      exactactiveclass="router-link-exact-active"
                      routertag="a"
                      target="_self"
                      type="button"
                    >
                      Forgot password?
                    </cbutton-stub>
                     
                    <cbutton-stub
                      activeclass="router-link-active"
                      class="d-lg-none"
                      color="link"
                      event="click"
                      exactactiveclass="router-link-exact-active"
                      routertag="a"
                      target="_self"
                      type="button"
                    >
                      Register now!
                    </cbutton-stub>
                  </ccol-stub>
                </crow-stub>
              </cform-stub>
            </ccardbody-stub>
          </ccard-stub>
           
          <ccard-stub
            bodywrapper="true"
            class="text-center py-5 d-md-down-none"
            color="primary"
            textcolor="white"
          >
            <ccardbody-stub>
              <h2>
                Sign up
              </h2>
               
              <p>
                Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
              </p>
               
              <cbutton-stub
                activeclass="router-link-active"
                color="light"
                event="click"
                exactactiveclass="router-link-exact-active"
                routertag="a"
                size="lg"
                target="_self"
                type="button"
                variant="outline"
              >
                
                Register Now!
              
              </cbutton-stub>
            </ccardbody-stub>
          </ccard-stub>
        </ccardgroup-stub>
      </ccol-stub>
    </crow-stub>
  </ccontainer-stub>
</div>
`;
